lib/yargs-factory.ts

```diff
@@ -904,6 +904,21 @@ export class YargsInstance {
         opt = {};
       }
 
+      // Prevent version name collision
+      // Addresses: https://github.com/yargs/yargs/issues/1979
+      if (this.#versionOpt && (key === 'version' || opt?.alias === 'version')) {
+        throw new YError(
+          [
+            '"version" is a reserved word.',
+            'Please do one of the following:',
```

Comment:
I think throwing an error is too heavy handed, as it would break a lot of applications. Here's what I would do:

1. emit a custom warning with a similar message to this, using the approach described [here](https://nodejs.org/api/process.html#process_emitting_custom_warnings). We should make sure that we don't show it more than once, using the approach outlined in the docs.
2. I think we should turn off our own handling of `.version()` if someone configures it themselves, which the warning message can draw attention to.

What I'm not sure of, is whether we should call this a breaking change.

Reply:
A couple of questions:
- I want to capture output and prevent it from going to stdout. Is patching `process.emitWarning` in `checkUsage` the way to go? I see that there are already patches for a few process and console methods there.
- Would emitting a warning be sufficient? It would keep this from becoming a breaking change.